assets/src/realtime.ts

```diff
@@ -53,26 +53,59 @@ export interface Ghost {
 
 export type NotificationId = string
 
-export interface Notification {
-  id: NotificationId
+// Matches the Module in which this notification was created by
```

Comment:
Interesting! How does this work? This seems a little magical to me, how the Elixir module is referenced by this TS file. This change is the foundation of the refactor (I see NotificationType is referenced throughout the repo), so having some breadcrumbs for future readers could be helpful

Reply:
These just happen to be the module name because that's what I set the `default:` to. https://github.com/mbta/skate/pull/2773/files#diff-226c071046feb083bddbfbfd9ff74180d55bb17a2a0b27a1e1b423b59ddde1f9R40

We could use a different value but I didn't think it was worth agonizing over when I was in the middle of this, and I like that this is "self documenting" in that it points you to which module it's from.

I'll think about how to document this 👍🏻 

I did try and think of an alternative to adding a `$type` virtual key to the ecto schema, but I didn't think it was worth creating a new struct type just for that, and I didn't want to write a custom `Jason.Encode` implementation _just_ to add a type discriminator for typescript. This way, ecto just puts the key on the struct for us, but maybe this isn't a good pattern?

Reply:
I added two commits which change the key on the Elixir side from `$type` to `__struct__`, which I help is a bit more explanatory in that it doesn't require knowing that the struct would contain a `$type` key with the `__MODULE__` but would instead just reference the actual Elixir struct we expect to receive. Do you think this makes this any better?

See commits beginning with `review!`, e.g.,

[review! fix: convert $type in notifications to __struct__](https://github.com/mbta/skate/pull/2773/commits/f3fedf8e1c0c504a9a28fe146bcd49a724b76dd5)

Reply:
I like it! Thank you